File: swift/problem172.swift
class Solution {
    func trailingZeroes(_ n: Int) -> Int {
        if (n < 1) {
            return 0
        }
        
        var n = n; var ans = 0
        
        while (n != 0) {
            ans += n/5
            n /= 5
        }
        
        return ans
    }
}

File: java/problem172.java
class Solution {
    /*
    less than 5----->0;
    [5,10)----->1;

    special:5^{x} log5(n)

    total num of factor_5
    */
    public int trailingZeroes(int n) {
        long temp=5;
        int result=n/5;
        while(temp<=n){
            temp*=5;
            result+=n/temp;
        }
        return result;
    }
}
public class problem172{
     public static void main(String args[]){
         Solution test=new Solution();
         
         System.out.println(test.trailingZeroes(200));
    }
}